src/lib/mailer/index.js

```diff
@@ -101,11 +101,18 @@ export const onFailedNotification = (job, error) => {
 };
 
 export const onCompletedNotification = (job, result) => {
-  if (result != null) {
-    logger.info(`Successfully sent ${job.name} notification for ${job.data.report.displayId || job.data.report.id}`);
-    logEmailNotification(job, true, result);
+  if (job.data.reports && Array.isArray(job.data.reports)) {
+    job.data.reports.forEach((report) => {
+      if (result != null) {
```

Comment:
Looks like this assumes if we have a result then we will have a report.displayId? Should the if statment check anything that we are accessing here in the loop?